app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt

```diff
@@ -226,13 +236,34 @@ class ChatRoomFragment : Fragment(), ChatRoomView, EmojiKeyboardListener, EmojiR
     }
 
     override fun onActivityResult(requestCode: Int, resultCode: Int, resultData: Intent?) {
-        if (requestCode == REQUEST_CODE_FOR_PERFORM_SAF && resultCode == Activity.RESULT_OK) {
-            if (resultData != null) {
-                uploadFile(resultData.data)
+        if (resultData != null && resultCode == Activity.RESULT_OK) {
+            when(requestCode){
+                REQUEST_CODE_FOR_PERFORM_SAF -> {
+                    uploadFile(resultData.data)
+                }
+                REQUEST_CODE_FOR_DRAW -> {
+                    val result= resultData.getByteArrayExtra("bitmap")
+                    val bitmap = BitmapFactory.decodeByteArray(result, 0, result.size)
+                    val uri = saveImage(bitmap)
+                    uploadFile(uri)
+                }
             }
         }
     }
 
+    private fun saveImage(bitmap: Bitmap): Uri {
```

Comment:
Actually this create a path to salve the image, compress the image and return an URI.
Could you change for meaningful methods?